src/myrtlespeech/builders/task_config.py

```diff
@@ -116,24 +116,34 @@ def target_transform(target):
         add_seq_len_to_transforms=True,
     )
 
-    shuffle = task_config.train_config.shuffle_batches_before_every_epoch
+    shuffle_str = task_config.train_config.WhichOneof("shuffle_strategy")
     batch_sampler: Union[SortaGrad, SequentialRandomSampler]
-    if task_config.train_config.sortagrad:
+    if shuffle_str == "sortagrad":
         batch_sampler = SortaGrad(
             indices=range(len(train_dataset)),
             batch_size=task_config.train_config.batch_size,
-            shuffle=shuffle,
+            shuffle=True,
             drop_last=False,
         )
         sort = True
-    else:
+    elif shuffle_str == "random_batches":
+        batch_sampler = SequentialRandomSampler(
+            indices=range(len(train_dataset)),
+            batch_size=task_config.train_config.batch_size,
+            shuffle=True,
+            drop_last=False,
+        )
+        sort = False
```

Comment:
Does sorting within a single batch matter? Why does `SortaGrad` require it and the other two cases not?

Reply:
What I thought was that it could matter especially in the case you use big batch sizes. Moreover I saw that in deepspeech_internal the collate function also sorts every single batch, so I sticked with that implementation. 
The value of the `sort` variable is only used inside the collate function and it is True only when we want to sort every single batch. It is set to false in the `sequential_batches` if case because I have interpreted it as "we want to go through every batch in a sequential way but we don't care about sorting single batches". 
On the other hand I set it to true in the `sorta_grad`case because I have interpreted it as "we want to go through every batch in a sequential way for the first epoch and we also care about sorting each single batch". 
Let me know if my interpretation was wrong and something should be changed.

Reply:
What would change when the batch size increases?

The order of samples within a batch has no effect on the training process unless I'm overlooking something? i.e. the loss, gradient, etc will be the same.

If the above is true sort should just be fixed to one value for all cases to simplify the logic? True?

Reply:
Make sense, I interpreted wrongly the sorting inside the collate function in deepspeech_internal. I have just removed the sort variable and changed consequently the collate function. 
What are the init parameters for which I should add the documentation?